plugins/woocommerce/client/admin/client/utils/admin-settings.js

```diff
@@ -38,6 +38,17 @@ export function getAdminSetting(
 	fallback = false,
 	filter = ( val ) => val
 ) {
+	if ( window.deprecatedWcSettings ) {
+		const key = `admin.${ name }`;
+		if ( key in window.deprecatedWcSettings ) {
+			const warningMessage = window.deprecatedWcSettings[ key ]
+				? window.deprecatedWcSettings[ key ]
+				: `Deprecated: ${ name } has been deprecated. This value will be removed soon.`;
```

Comment:
I think it would be great if we could mention the alternative in the deprecated message and maybe be more specific about which version it will be removed. What do you think?

Reply:
@chihsuan So this is the default message. We can work on a customized message with an alternative when we work on a PR to add more properties other than admin.onboarding

> maybe be more specific about which version it will be removed
Hey @adrianduffell ✋ How long should we keep the deprecation messages? Do you think two release cycles would be long enough?

Reply:
@moon0326 , yeah let's do the removal in the 10.0 release.

Reply:
@adrianduffell Sounds good! Updated messages in https://github.com/woocommerce/woocommerce/pull/55707/files#diff-b5ccd801e97e593229d243c9c5721c56d282ac1e2867df738e62d4ea105bfcbfR25